tests/test_csrc/CMakeLists.txt

```diff
@@ -14,60 +14,64 @@ aux_source_directory(${CMAKE_CURRENT_SOURCE_DIR}/net NET_TC)
-else()
-  set(TASK_LIST "model")
-  set(CODEBASES "${MMDEPLOY_CODEBASES}")
-  if("mmcls" IN_LIST MMDEPLOY_CODEBASES)
-    list(APPEND TASK_LIST "classifier")
-  endif()
-  if("mmdet" IN_LIST MMDEPLOY_CODEBASES)
-    list(APPEND TASK_LIST "detector")
-  endif()
-  if("mmseg" IN_LIST MMDEPLOY_CODEBASES)
-    list(APPEND TASK_LIST "segmentor")
-  endif()
-  if("mmedit" IN_LIST MMDEPLOY_CODEBASES)
-    list(APPEND TASK_LIST "restorer")
-  endif()
-  if("mmocr" IN_LIST MMDEPLOY_CODEBASES)
-    list(APPEND TASK_LIST "text_detector")
-    list(APPEND TASK_LIST "text_recognizer")
-  endif()
-endif()
-foreach(TASK ${TASK_LIST})
-  list(APPEND CAPI_TC ${CMAKE_CURRENT_SOURCE_DIR}/capi/test_${TASK}.cpp)
-endforeach()
+if ("all" IN_LIST MMDEPLOY_CODEBASES)
+    set(TASK_LIST
+            "classifier;detector;segmentor;text_detector;text_recognizer;restorer;model"
+            )
+    set(CODEBASES "mmcls;mmdet;mmseg;mmedit;mmocr")
+else ()
+    set(TASK_LIST "model")
+    set(CODEBASES "${MMDEPLOY_CODEBASES}")
+    if ("mmcls" IN_LIST MMDEPLOY_CODEBASES)
+        list(APPEND TASK_LIST "classifier")
+    endif ()
+    if ("mmdet" IN_LIST MMDEPLOY_CODEBASES)
+        list(APPEND TASK_LIST "detector")
+    endif ()
+    if ("mmseg" IN_LIST MMDEPLOY_CODEBASES)
+        list(APPEND TASK_LIST "segmentor")
+    endif ()
+    if ("mmedit" IN_LIST MMDEPLOY_CODEBASES)
+        list(APPEND TASK_LIST "restorer")
+    endif ()
+    if ("mmocr" IN_LIST MMDEPLOY_CODEBASES)
+        list(APPEND TASK_LIST "text_detector")
+        list(APPEND TASK_LIST "text_recognizer")
+    endif ()
+endif ()
+foreach (TASK ${TASK_LIST})
+    list(APPEND CAPI_TC ${CMAKE_CURRENT_SOURCE_DIR}/capi/test_${TASK}.cpp)
+endforeach ()
 
 # generate the header file
 configure_file(config/test_define.h.in
-               ${CMAKE_CURRENT_SOURCE_DIR}/test_define.h)
+        ${CMAKE_CURRENT_SOURCE_DIR}/test_define.h)
 
 set(TC_SRCS
-    ${TC_SRCS}
-    ${ARCHIVE_TC}
-    ${CORE_TC}
-    ${TRANSFORM_TC}
-    ${MODEL_TC}
-    ${NET_TC}
-    ${DEVICE_TC}
-    ${CAPI_TC})
+        ${TC_SRCS}
+        ${ARCHIVE_TC}
+        ${CORE_TC}
+        ${TRANSFORM_TC}
+        ${MODEL_TC}
+        ${NET_TC}
+        ${DEVICE_TC}
+        ${CAPI_TC})
 
 add_executable(mmdeploy_tests ${TC_SRCS})
 target_include_directories(mmdeploy_tests
-                           PRIVATE ${CMAKE_SOURCE_DIR}/third_party/catch2)
+        PRIVATE ${CMAKE_SOURCE_DIR}/third_party/catch2)
 target_include_directories(mmdeploy_tests PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
-target_link_libraries(
-  mmdeploy_tests PRIVATE ${MMDEPLOY_LIBS} ${OpenCV_LIBS}
-        -Wl,--no-as-needed ${MMDEPLOY_DYNAMIC_MODULES} -Wl,--as-need
-        -Wl,--whole-archive ${MMDEPLOY_STATIC_MODULES} -Wl,--no-whole-archive)
+
+mmdeploy_load_static(mmdeploy_tests MMDeployStaticModules)
+mmdeploy_load_dynamic(mmdeploy_tests MMDeployDynamicModules)
+target_link_libraries(mmdeploy_tests PRIVATE
+        MMDeployLibs
+        mmdeploy_core
```

Comment:
It seems no need to link `mmdeploy_core`, `mmdeploy_transform` or `mmdeploy_open_utils` any more

Reply:
Can't do that. Exported symbols from `mmdeploy_transform` and `mmdeploy_opencv_utils` are used in preprocess TCs. They are needed when building shared libs.

Actually, only `mmdeploy_core` may be removed here because it's included in `MMDeployLibs`.